pkg/v1/mutate/rebase.go

```diff
@@ -142,3 +143,16 @@ func createAddendums(startHistory, startLayer int, history []v1.History, layers
 
 	return adds
 }
+
+func Manifest(base v1.Image, m *v1.Manifest) (v1.Image, error) {
```

Comment:
I think what you _really_ want is `mutate.Annotations` or something? Allowing you to set the whole manifest introduces some weirdness around invariant violations (`mutate.ConfigFile` does too, but I'd like to minimize the amount of it).

Reply:
(Ideally we could make this work for `Image`, `ImageIndex`, and `Layer`, but I'm fine with three variants of that to make the type system nonsense a little less silly.)

Reply:
Done, with `mutate.Annotations` (it only supports `v1.Image`, maybe we should expand that?)